lib/resources/forms.js

```diff
@@ -136,6 +136,18 @@ module.exports = (service, endpoint) => {
       .then(((form) => Promise.all([ Forms.publish(form), Submissions.clearDraftSubmissions(form.id) ])))
       .then(success)));
 
+  // Entity/Dataset-specific endpoint that is used to show how publishing
+  // a form will change any datasets mentioned in the form.
+  // Even though there are dataset-related permissions, we will use the form modification verbs
+  // for authentication because it seems weird to mix the ability/inability to alter a dataset
+  // with the ability to modify forms -- they should just be the same.
+  service.get('/projects/:projectId/forms/:id/draft/dataset-diff', endpoint(({ Forms, Datasets }, { params, auth }) =>
+    Forms.getByProjectAndXmlFormId(params.projectId, params.id, false, Form.DraftVersion)
+      .then(getOrNotFound)
+      .then((form) => auth.canOrReject('form.update', form)) // use form verb
+      .then(ensureFormDef)
+      .then(() => Datasets.getDatasetDiff(params.id))));
```

Comment:
Should this pass in the project ID? The `xmlFormId` won't be unique on its own.

Reply:
agree, I missed adding `projectId` in the query

Reply:
Probably that change should come in a separate PR, I just thought I'd leave a comment here as I noticed it.